lib/dialects/mysql/index.js

```diff
@@ -179,6 +180,7 @@ class Client_MySQL extends Client {
         if (conn.__knex__disposed) {
           this.logger.warn(`Connection Error: ${conn.__knex__disposed}`);
         }
+        connectionToKill.resume();
```

Comment:
minor tweak: this could be run before connection destroy, so working connection will continue to work earlier

Reply:
Moved in f1a3d8a